### PR TITLE
Migrate io.zeebe.broker.engine.incident tests

### LINKED ISSUE #2461: Move processing tests into engine module
During the creating of the engine module we were not able to move the most of the processing tests, because they most of them need an `EmbeddedBrokerRule`.

We could rewrite the test in a way such that they don't need to start a complete broker. After that we could move them into the corresponding module. This would decrease the test execution time.

### LINKED ISSUE #2555: Migrate io.zeebe.broker.engine.incident.* tests
Migrate all tests which are part of the package `io.zeebe.broker.engine.incident` to the engine module and move them in a corresponding package, related to the processor.

blocks #2461

### PR DESCRIPTION
**Pull request by ChrisKujawa**
closes #2555 
closes #2461

### CODE REVIEW

File: engine/src/test/java/io/zeebe/engine/processor/workflow/incident/MappingIncidentTest.java [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+        .hasErrorMessage("No data found for query foo.")
+        .hasBpmnProcessId("process")
+        .hasWorkflowInstanceKey(workflowInstanceKey)
+        .hasElementId("failingTask")
+        .hasElementInstanceKey(incidentResolvedEvent.getValue().getElementInstanceKey())
+        .hasVariableScopeKey(incidentResolvedEvent.getValue().getElementInstanceKey());
+  }
+
+  @Test
+  @Category(UnstableTest.class)
```
Comment (MiguelPires) [inline engine/src/test/java/io/zeebe/engine/processor/workflow/incident/MappingIncidentTest.java]: Is this still necessary? I'd assume that the previous flakiness is gone, I also couldn't get it to fail locally

**Review by miguelpires — COMMENTED** [review]
Just a small question

File: engine/src/test/java/io/zeebe/engine/processor/workflow/incident/MappingIncidentTest.java (hunk 1 above)
Comment (Zelldon) [reply engine/src/test/java/io/zeebe/engine/processor/workflow/incident/MappingIncidentTest.java]: I think you're right we can remove it nice catch

**Comment by Zelldon** [comment]
@MiguelPires I removed the annotation

**Review by miguelpires — APPROVED** [review]

**Pull request merged**